### PR TITLE
Add support for the -impredicative-set option

### PR DESCRIPTION
**Pull request by dhilst**
Closes https://github.com/ejgallego/coq-serapi/issues/287

### CODE REVIEW

**Comment by dhilst** [comment]
Hello @ejgallego 

I think I have the patch that is needed for allowing `-impredicative-set`. I built the `sercomp` locally but when I run I got this error, I'm not sure what is going on, or how to debug this (opam/dune is still a new thing to me)

```
➜  coq-serapi git:(v8.16) _build/install/default/bin/sercomp -Q ../coq/proj/src,MyProj  src/Bar.v
[sertop] Critical Dynlink error implementation mismatch on Serlib__Ser_names
Error: 
       Dynlink error: implementation mismatch on Serlib__Ser_names
       Findlib paths:/home/geckos/.opam/coq-serapi/lib
```

**Comment by ejgallego** [comment]
That's a very confusing error indeed; there is a bit of explanation on `notes/build.md`, but indeed, that error is due to `OCAMLPATH` not being setup properly, then `findlib` picks the wrong library.

Easiest fix is to use `dune exec -- sertop` which will take care of setting `OCAMLPATH` for you.

File: sertop/sertop_init.ml [hunk 1]
```diff
@@ -59,7 +62,7 @@ let init_runtime opts =
   Mltop.init_known_plugins ();
 
   (* This should be configurable somehow. *)
```
Comment (ejgallego) [inline sertop/sertop_init.ml:64]: Comment is outdated now.

**Review by ejgallego — APPROVED** [review]
LGTM, can you add an entry in `CHANGES.md` ?

**Comment by dhilst** [comment]
> LGTM, can you add an entry in `CHANGES.md` ?

Done, and I also updated the comment above

**Comment by dhilst** [comment]
A question that I have, I'm using Coq 8.14, to port this to the `serapi` 8.14 I should create another MR backporting this MR and pointing it to the 8.14 branch right?

**Comment by ejgallego** [comment]
> > LGTM, can you add an entry in `CHANGES.md` ?
> 
> Done, and I also updated the comment above

Thanks! This will go in 8.16.2 but I can fix this direclty on the tree.

> I'm using Coq 8.14, to port this to the serapi 8.14 I should create another MR backporting this MR and pointing it to the 8.14 branch right?

Yes, that should work fine.

**Comment by dhilst** [comment]
There is a build failing, I will take a look

**Review by ejgallego — APPROVED** [review]

**Pull request merged**